src/stores/OriginalContractsStore.ts

```diff
@@ -1,24 +1,61 @@
 import { proxy } from 'valtio'
 import { subscribeKey } from 'valtio/utils'
 import WalletStore from 'stores/WalletStore'
+import defaultProvider from 'helpers/defaultProvider'
 import getOwnedERC721 from 'helpers/getOwnedERC721'
 
 interface ContractsStoreType {
-  contractsOwned: Promise<string[]>
+  contractsOwnedState: { [token: string]: number }
+  contractsOwnedStateAccount?: string
+  contractsOwned: string[]
+  lastBlockId?: number
+  locked: boolean
   fetchContractsOwned: () => void
 }
 
 const OriginalContractsStore = proxy<ContractsStoreType>({
-  contractsOwned: Promise.resolve([]),
-  fetchContractsOwned() {
-    OriginalContractsStore.contractsOwned = WalletStore.account
-      ? getOwnedERC721(WalletStore.account)
-      : Promise.resolve([])
+  locked: false,
+  contractsOwnedState: {},
+  contractsOwned: [],
+  async fetchContractsOwned() {
+    if (
+      OriginalContractsStore.contractsOwnedStateAccount !== WalletStore.account
```

Comment:
Should this happen in the `account` listener?

Reply:
Maybe I misunderstood, but checking the selected account should include both listeners